pkg/controller/migmigration/task.go

```diff
@@ -1050,6 +1055,9 @@ func (t *Task) updatePipeline() {
 	if currentStep != nil {
 		currentStep.MarkStarted()
 		currentStep.Phase = t.Phase
+		if currentStep.Name == StepDirectVolume {
```

Comment:
as pointed in the previous comment, let's not do this for one particular step, instead introduce a function to set custom phase description.

Reply:
since initPipeline() is called at the start of the loop, you can always call the setDescription() function to override the empty string.

Reply:
```
From 4e0f75ae4b392f2ca001d47418c67b5cff958021 Mon Sep 17 00:00:00 2001
From: Pranav Gaikwad <pgaikwad@redhat.com>
Date: Tue, 24 Nov 2020 13:45:32 -0500
Subject: [PATCH] introduce function to set step description

---
 pkg/controller/migmigration/task.go | 23 +++++++++++------------
 1 file changed, 11 insertions(+), 12 deletions(-)

diff --git a/pkg/controller/migmigration/task.go b/pkg/controller/migmigration/task.go
index c5cbd1ed..be77eb34 100644
--- a/pkg/controller/migmigration/task.go
+++ b/pkg/controller/migmigration/task.go
@@ -649,21 +649,18 @@ func (t *Task) Run() error {
 				return liberr.Wrap(err)
 			}
 		}
-		t.Owner.Status.FindStep(t.Step).Message = dvm.Status.PhaseDescription
 		// Check if DVM is complete and report progress
 		completed, reasons, progress := t.hasDirectVolumeMigrationCompleted(dvm)
+		t.setStepDescription(dvm.Status.PhaseDescription)
+		t.setProgress(progress)
 		if completed {
-			step := t.Owner.Status.FindStep(t.Step)
-			step.MarkCompleted()
 			if len(reasons) > 0 {
 				t.setDirectVolumeMigrationFailureWarning(dvm)
 			}
-			t.setProgress(progress)
 			if err = t.next(); err != nil {
 				return liberr.Wrap(err)
 			}
 		} else {
-			t.setProgress(progress)
 			t.Requeue = PollReQ
 		}
 	case EnsureStageBackup:
@@ -1042,8 +1039,6 @@ func (t *Task) updatePipeline() {
 		}
 	}
 
-	defer t.Owner.Status.ReflectPipeline()
-
 	// mark steps skipped
 	for _, step := range t.Owner.Status.Pipeline {
 		if step == currentStep {
@@ -1055,18 +1050,22 @@ func (t *Task) updatePipeline() {
 	if currentStep != nil {
 		currentStep.MarkStarted()
 		currentStep.Phase = t.Phase
-		if currentStep.Name == StepDirectVolume {
-			return
-		}
 		if desc, found := PhaseDescriptions[t.Phase]; found {
 			currentStep.Message = desc
-		} else {
-			currentStep.Message = ""
 		}
 		if t.Phase == Completed {
 			currentStep.MarkCompleted()
 		}
 	}
+	t.Owner.Status.ReflectPipeline()
+}
+
+// setStepDescription overrides default phase description with custom string
+func (t *Task) setStepDescription(desc string) {
+	currentStep := t.Owner.Status.FindStep(t.Step)
+	if currentStep != nil {
+		currentStep.Message = desc
+	}
 }
 
 func (t *Task) setProgress(progress []string) {
-- 
2.26.2

```